### DISCUSSION TITLE
Oqtane 3.2 Released

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#2420] [General]
Release announcement here: https://www.oqtane.org/blog/!/50/oqtane-3-2-for--net-maui---blazor-hybrid

**Comment by lanthonyneville** [comment]
Hi @sbwalker - can you clarify something about the usage of the runtime hosting models ....?
Essentially my question is - is the choice of hosting model set globally for all modules/pages in the application, or can it vary? I am pretty sure it is necessarily global and it is set in the appsettings.json file via the Runtime setting .... is that right? I found an article that indicated that it is possible to switch models at runtime, but not sure if Oqtane could or would support this (https://itnext.io/blazor-switching-server-and-webassembly-at-runtime-d65c25fd4d8).
My use case is this - most modules in my application run well under the Server hosting model, however there may be some that would work better as WebAssembly or MAUI. I am wondering whether I need to run these under a separate application instance, with some kind of navigation between the "server-side" site and the "client-side" site.
Thanks

**Reply by sbwalker** [reply]
@lanthonyneville In version 3.0 Oqtane was enhanced to allow you to specify the hosting model for each site in your installation. So the hosting model can vary by site (the old appsettings.json approach is no longer used) and is determined dynamically at runtime. In theory it would be possible to set the hosting model for a specific page, however it would require a server-side navigation link so that it could load all of the modules and register the services on the client - so my assumption would be that the user experience would be rather poor because the transition would not be seamless. It is important to realize that the "hosting model" is more of a client-side concept than a server-side concept. For example, a MAUI app is a dedicated client application - there is no hosting model setting required on the server at all as the MAUI client app simply treats it as a standard API for retrieving data. WebAssembly is also a dedicated client application - however in Oqtane we are using the ASP.NET Hosted option rather than Stand-Alone so the framework loads the _host.cshtml page on the server (rather than using its own index.html) and this is the only reason why a hosting model specification is required. I would not support the approach described in the link you posted of loading Server and WebAssembly concurrently as this is not an officially supported use case by Microsoft and would likely have unexpected side effects based on the fact that you are breaking the separation of concerns principle.

**Reply by lanthonyneville** [reply]
Thanks for the thorough reply. I missed that change in 3.0 - could be useful!

**Reply by lanthonyneville** [reply]
Just an FYI - there is a proposal for "mixed" Blazor Server and WebAssembly here: https://github.com/dotnet/aspnetcore/issues/38128 ... it sounds like it is under consideration by MS but "moved to a .Net 8 milestone".
